src/Client.php

```diff
@@ -1127,22 +1127,33 @@ public function getLearnerReport(GetLearnerReportQuery $query): array {
             }
             if (isset($report->VariantEndDate)) {
                 $currentReport->setVariantEndDate(
-                    new DateTime((string) $report->VariantEndDate)
+                    $this->setDateToNullOrDateTimeObject((string) $report->VariantEndDate)
                 );
             }
             if (isset($report->VariantName)) {
                 $currentReport->setVariantName((string) $report->VariantName);
             }
             if (isset($report->VariantStartDate)) {
                 $currentReport->setVariantStartDate(
-                    new DateTime((string) $report->VariantStartDate)
+                    $this->setDateToNullOrDateTimeObject((string) $report->VariantStartDate)
                 );
             }
             $learnerReports[] = $currentReport;
         }
 
         return $learnerReports;
     }
+    
+    /**
+     * Sets the provided date to either null or a DateTime object.
+     */
+    private function setDateToNullOrDateTimeObject(string $dateString): ?DateTime {
```

Comment:
- First, I find this method name to be misleading... you are not setting a property of anything you are decoding a nullable datetime. So I think a better name would be `toDateTimeInterface`. 
- Second, I wish you declared that you returned a `DateTimeInterface|null` and you actually returned a `DateTimeImmutable`. This prevents accidental modification to properties of data types.
- Third, Client is starting to get less organized. We have a few other private methods: `logFailedRequest` and `getErrorCodesFromXmlElement` which are furtherdown and it feels like this method should be in a region with `getErrorCodesFromXmlElement` at least as it is a decoding utility method if not a region of private methods.